### PR TITLE
PR for Server Calling SDK Public Preview

### PR DESCRIPTION
**Pull request by arifsaikat-microsoft**
PR for Server Calling SDK Public Preview

### CODE REVIEW

**Review by zhengni-msft — APPROVED** [review]

**Review by zhengni-msft — COMMENTED** [review]
Will it for more changes with swagger review results? Change to comment status.

**Review by chrwhit — APPROVED** [review]

**Comment by arifsaikat-microsoft** [comment]
> Will it for more changes with swagger review results? Change to comment status.

Yes, these changes are mainly name changes and few contract updates based on api review feedback

File: sdk/communication/azure-communication-callingserver/src/test/java/com/azure/communication/callingserver/CallConnectionAsyncLiveTests.java [hunk 1]
```diff
@@ -27,6 +28,14 @@
...
     @MethodSource("com.azure.core.test.TestBase#getHttpClients")
     public void runCreatePlayCancelHangupScenarioAsync(HttpClient httpClient) {
+
+        // This test requires human intervention to record and
+        // will not function in live mode.
+        if (this.getTestMode() == TestMode.LIVE) {
+            System.out.println("Warning: Test is skipped, does not support live mode.");
+            return;
+        }
+
```
Comment (JianpingChen) [inline sdk/communication/azure-communication-callingserver/src/test/java/com/azure/communication/callingserver/CallConnectionAsyncLiveTests.java:38]: Consider using a pattern similar to this.

@DisabledIfEnvironmentVariable(
        named = "COMMUNICATION_SKIP_INT_PHONENUMBERS_TEST",
        matches = "(?i)(true)")

c.f sdk/communication/azure-communication-phonenumbers/src/test/java/com/azure/communication/phonenumbers/PhoneNumbersAsyncClientIntegrationTest.java#L90

Comment (arifsaikat-microsoft) [reply sdk/communication/azure-communication-callingserver/src/test/java/com/azure/communication/callingserver/CallConnectionAsyncLiveTests.java]: this pattern is applied here and other related places

File: sdk/communication/azure-communication-callingserver/src/test/java/com/azure/communication/callingserver/CallConnectionAsyncLiveTests.java [hunk 2]
```diff
@@ -27,15 +27,10 @@
 
     @ParameterizedTest
     @MethodSource("com.azure.core.test.TestBase#getHttpClients")
+    @DisabledIfEnvironmentVariable(
+        named = "COMMUNICATION_SKIP_INT_CALLINGSERVER_TEST",
```
Comment (JianpingChen) [inline sdk/communication/azure-communication-callingserver/src/test/java/com/azure/communication/callingserver/CallConnectionAsyncLiveTests.java:31]: Maybe a better name. This is not for INT only.

**Review by JianpingChen — APPROVED** [review]

File: sdk/communication/azure-communication-callingserver/src/test/java/com/azure/communication/callingserver/CallConnectionAsyncLiveTests.java (hunk 2 above)
Comment (JianpingChen) [reply sdk/communication/azure-communication-callingserver/src/test/java/com/azure/communication/callingserver/CallConnectionAsyncLiveTests.java:31]: INT was used for **INT environment** not **Integration tests.** :-)

File: sdk/communication/azure-communication-callingserver/src/test/java/com/azure/communication/callingserver/CallConnectionAsyncLiveTests.java [hunk 3]
```diff
@@ -2,30 +2,34 @@
...
 public class CallConnectionAsyncLiveTests extends CallingServerTestBase {
 
     private final String fromUser = getNewUserId();
     private final String toUser = getNewUserId();
 
     @ParameterizedTest
     @MethodSource("com.azure.core.test.TestBase#getHttpClients")
+    @DisabledIfEnvironmentVariable(
+        named = "COMMUNICATION_SKIP_INT_CALLINGSERVER_TEST",
+        matches = "(?i)(true)")
```
Comment (JianpingChen) [inline sdk/communication/azure-communication-callingserver/src/test/java/com/azure/communication/callingserver/CallConnectionAsyncLiveTests.java:32]: FYI: It is also possible to add a comment for reason, disabledReason="Require human intervention"

File: sdk/communication/azure-communication-callingserver/src/test/java/com/azure/communication/callingserver/CallConnectionAsyncLiveTests.java (hunk 2 above)
Comment (arifsaikat-microsoft) [reply sdk/communication/azure-communication-callingserver/src/test/java/com/azure/communication/callingserver/CallConnectionAsyncLiveTests.java]: env variable renamed to RUN_CALLINGSERVER_TEST_RECORD

File: sdk/communication/azure-communication-callingserver/src/test/java/com/azure/communication/callingserver/CallConnectionAsyncLiveTests.java (hunk 3 above)
Comment (arifsaikat-microsoft) [reply sdk/communication/azure-communication-callingserver/src/test/java/com/azure/communication/callingserver/CallConnectionAsyncLiveTests.java]: disable reason added

    @EnabledIfEnvironmentVariable(
        named = "RUN_CALLINGSERVER_TEST_RECORD",
        matches = "(?i)(true)",
        disabledReason = "Requires human intervention")

File: sdk/communication/azure-communication-callingserver/pom.xml [hunk 4]
```diff
@@ -42,7 +42,7 @@
     <test.dir>src/test</test.dir>
     <jacoco.min.linecoverage>0.70</jacoco.min.linecoverage>
     <jacoco.min.branchcoverage>0.70</jacoco.min.branchcoverage>
-    <jacoco.skip.coverage.check>true</jacoco.skip.coverage.check>
+    <jacoco.skip.coverage.check>false</jacoco.skip.coverage.check>
```
Comment (JianpingChen) [inline sdk/communication/azure-communication-callingserver/pom.xml:45]: This is where the build failed. Either disable for now (true) or lower the coverage until it is raised

Comment (arifsaikat-microsoft) [reply sdk/communication/azure-communication-callingserver/pom.xml:45]: using @DisabledIfEnvironmentVariable to disable some tests from LIVE tests, but they will be part of PLAYBACK, so coverage should be up now.

Comment (arifsaikat-microsoft) [reply sdk/communication/azure-communication-callingserver/pom.xml:45]: Will revisit when these disabled tests are ready to be part of LIVE tests

**Review by minnieliu — APPROVED** [review]

**Pull request merged**